package_managers/install_pkgs.bzl

```diff
@@ -0,0 +1,94 @@
+#Copyright 2017 Google Inc. All rights reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Rule for installing apt packages from a tar file into a docker image."""
+
+def _impl(ctx):
+  builder_image_name = "bazel/%s:%s" % (ctx.attr.image_tar.label.package,
+                                        ctx.attr.image_tar.label.name.split(".tar")[0])
+  unstripped_tar = ctx.actions.declare_file(ctx.outputs.out.basename + ".unstripped")
+
+  build_contents = """\
+#!/bin/bash
+set -ex
+find .
+docker load --input {image_tar}
+
+cid=$(docker run -d -v $(pwd)/{installables}:/tmp/installables.tar -v $(pwd)/{installer}:/tmp/installer.sh --privileged {image_name} /tmp/installer.sh)
+
+docker attach $cid
+docker commit -c 'CMD /bin/bash' $cid {output_image_name}
+docker save {output_image_name} > {output_file_name}
+""".format(image_tar=ctx.file.image_tar.path,
+           installables=ctx.file.installables_tar.path,
+           output_file_name=unstripped_tar.path,
+           installer=ctx.file._installer_script.path,
+           image_name=builder_image_name,
+           output_image_name=ctx.attr.output_image_name
+  )
```

Comment:
just personal preference, but because this is longer than one or two lines I think it looks a lot cleaner if you move this to a `install_packages.sh.tpl` file, then load it in as a template and use `ctx.actions.expand_template` to substitute in the necessary values. see [here](https://github.com/bazelbuild/rules_docker/blob/master/contrib/test.bzl#L39) for an example.